tests/unit/interpreter/interpreter_test_enhance.cc

```diff
@@ -0,0 +1,481 @@
+             wasm_runtime_deinstantiate(module_inst);
+         }
+         if (module) {
+             wasm_runtime_unload(module);
+         }
+         if (wasm_file_buf) {
+             delete[] wasm_file_buf;
+         }
+         wasm_runtime_destroy();
+     }
+ 
+     void load_wasm_file()
+     {
+         std::ifstream wasm_file(WASM_FILE, std::ios::binary);
+         ASSERT_TRUE(wasm_file.is_open()) << "Failed to open WASM file: " << WASM_FILE;
+         
+         std::vector<uint8_t> buffer(std::istreambuf_iterator<char>(wasm_file), {});
+         wasm_file_size = buffer.size();
+         wasm_file_buf = new unsigned char[wasm_file_size];
+         std::copy(buffer.begin(), buffer.end(), wasm_file_buf);
+ 
+         module = wasm_runtime_load(wasm_file_buf, wasm_file_size, error_buf,
+                                    sizeof(error_buf));
+         ASSERT_NE(module, nullptr) << "Load module failed: " << error_buf;
+     }
+ 
+     void instantiate_module()
+     {
+         module_inst = wasm_runtime_instantiate(
+             module, STACK_SIZE, HEAP_SIZE, error_buf, sizeof(error_buf));
+         ASSERT_NE(module_inst, nullptr) << "Instantiate module failed: " << error_buf;
+ 
+         exec_env = wasm_runtime_create_exec_env(module_inst, STACK_SIZE);
+         ASSERT_NE(exec_env, nullptr);
+     }
+ 
+     RuntimeInitArgs init_args;
+     wasm_module_t module = nullptr;
+     wasm_module_inst_t module_inst = nullptr;
+     wasm_exec_env_t exec_env = nullptr;
+     unsigned char *wasm_file_buf = nullptr;
+     uint32_t wasm_file_size = 0;
+     char error_buf[128] = { 0 };
+ };
+ 
+ /**
+  * Test Function 1: call_indirect() - Valid function call
+  * Target: core/iwasm/interpreter/wasm_runtime.c:call_indirect()
+  * Expected Coverage: ~15 lines (success path)
+  */
+ TEST_F(FunctionInvocationTest, CallIndirect_ValidFunction_ReturnsCorrectResult)
+ {
+     wasm_function_inst_t func = wasm_runtime_lookup_function(
+         module_inst, "test_call_indirect_valid");
+     ASSERT_NE(func, nullptr);
+ 
+     uint32_t wasm_argv[2];
+     wasm_argv[0] = 0;  // table index 0 (points to $add_func)
+     wasm_argv[1] = 15; // parameter value
+ 
+     bool success = wasm_runtime_call_wasm(exec_env, func, 2, wasm_argv);
+     ASSERT_TRUE(success);
+ 
+     // $add_func adds 10 to input: 15 + 10 = 25
+     ASSERT_EQ(wasm_argv[0], 25);
+ }
+ 
+ /**
+  * Test Function 1: call_indirect() - Invalid index
+  * Target: core/iwasm/interpreter/wasm_runtime.c:call_indirect()
+  * Expected Coverage: ~15 lines (error path)
+  */
+ TEST_F(FunctionInvocationTest, CallIndirect_InvalidIndex_FailsGracefully)
+ {
+     wasm_function_inst_t func = wasm_runtime_lookup_function(
+         module_inst, "test_call_indirect_invalid_index");
+     ASSERT_NE(func, nullptr);
+ 
+     uint32_t wasm_argv[1];
+     wasm_argv[0] = 42; // input value
+ 
+     bool success = wasm_runtime_call_wasm(exec_env, func, 1, wasm_argv);
+     ASSERT_FALSE(success);
```

Comment:
additionally, `wasm_runtime_get_exception()` should return a non-empty infomation.